### ISSUE TITLE
Establishing contributing guidelines

### ISSUE THREAD

**Issue by mumrah** [dpkp/kafka-python#93]
As part of my effort to make this project more community driven, and less _me_ driven, I'd like to establish some guidelines for contributing code.

Others, feel free to chime in.
## Code Style
- PEP8
- Don't be afraid of long variable/method names
- No meta-programming please (that crap gets confusing)
## Development

_This applies to folks who now have commit access_. Unless it's a minor/trivial feature, do the work on a branch and open a pull request. This increases visibility of changes and encourages peer review. Also Travis CI will build pull requests to make sure things look OK.
## Collaboration

Do everything publicly through GitHub unless there's a good reason not to. I've created a "discussion" and "announcement" label for issues - use those. For live discussion, use the Apache Kafka IRC channel on irc.freenode.net #kafka

Also, when possible, don't merge your own PRs. Or at least don't merge until someone has commented and given a +1.
## Releasing

Since this library will like be in flux as bugs are fixed, releases should happen fairly often. Up until now, people have been running master (presumably), but once things are published to PyPI there will need to be a release process. To keep things simple, let's just publish from stable master versions. These should be tagged with the version number.

The version number should follow the pattern: {kafka-version}-{python-version}, e.g. 0.8.1-1

Still not sure how to mitigate the whole 0.8.0 vs 0.8.1 thing.
## Testing

New functionality should bring new tests with it. Unit tests are nice to have, but the integration tests are ultimately more important.
